src/libs/actions/IOU.js

```diff
@@ -122,82 +134,273 @@ function createIOUTransaction(params) {
-                processIOUErrorResponse(response);
-                return;
-            }
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${groupChatReport.reportID}`,
+            value: null,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${groupChatReport.reportID}`,
+            value: {
+                0: {pendingAction: null},
+                [groupIOUReportAction.sequenceNumber]: {pendingAction: null},
+            },
+        },
+    ];
 
-            // This data needs to go from this:
-            // {reportIDList: [1, 2], chatReportIDList: [3, 4]}
-            // to this:
-            // [{reportID: 1, chatReportID: 3}, {reportID: 2, chatReportID: 4}]
-            // in order for getIOUReportsForNewTransaction to know which IOU reports are associated with which
-            // chat reports
-            const reportParams = [];
-            for (let i = 0; i < response.reportIDList.length; i++) {
-                reportParams.push({
-                    reportID: response.reportIDList[i],
-                    chatReportID: response.chatReportIDList[i],
-                });
-            }
-            getIOUReportsForNewTransaction(reportParams);
-            Navigation.navigate(ROUTES.getReportRoute(chatReportID));
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${groupChatReport.reportID}`,
+            value: {
+                pendingFields: {createChat: null},
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${groupChatReport.reportID}`,
+            value: {
+                0: {pendingAction: null},
+                [groupIOUReportAction.sequenceNumber]: {pendingAction: null},
+            },
+        },
+    ];
+
+    // Loop through participants creating individual chats, iouReports and reportActionIDs as needed
+    const splitAmount = IOUUtils.calculateAmount(participants, amount);
+    const splits = [{email: currentUserEmail, amount: IOUUtils.calculateAmount(participants, amount, true)}];
+
+    _.each(participants, (participant) => {
+        const email = OptionsListUtils.addSMSDomainIfPhoneNumber(participant.login).toLowerCase();
+        if (email === currentUserEmail) {
+            return;
+        }
+
+        const existingOneOnOneChatReport = ReportUtils.getChatByParticipants([email]);
+        const oneOnOneChatReport = existingOneOnOneChatReport || ReportUtils.buildOptimisticChatReport([email]);
+        let oneOnOneIOUReport;
+        if (oneOnOneChatReport.iouReportID) {
+            oneOnOneIOUReport = iouReports[`${ONYXKEYS.COLLECTION.REPORT_IOUS}${oneOnOneChatReport.iouReportID}`];
+            oneOnOneIOUReport.total += splitAmount;
```

Comment:
there will be one issue I faced with reject transaction too. I have left a comment here https://github.com/Expensify/App/pull/11720/files#r1002122139

Basically we can have a scenario, where User A is splitting money with users B, C and D. 

User B has already pending request for $4 from User A.

Now user A requested $32 between those 4 people so we split it equally and User A request $8 from each. Therefore now User B owes User A $4 but in the App this would not be reflected optimistically, the report total will show as $12 since the total is always positive (it is the manager and owner of the IOU report which flips)

We should make this into a helper method, would you like to incorporate this here or make sense to follow with the command like so CreateIOUTransaction -> RejectTransaction -> this command?

Reply:
Nae problem 🙇

Reply:
Updated with the new method!